### PR TITLE
Port stock_picking_invoice_link to the new API (and other improvements)

### PR DESCRIPTION
**Pull request by tafaRU**

### CODE REVIEW

File: stock_picking_invoice_link/README.rst [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+and see the pickings related to the invoice. This only applies
+on the invoices generated from pickings.
+
+Bug Tracker
+===========
+
+Bugs are tracked on `GitHub Issues <https://github.com/OCA/stock-logistics-workflow/issues>`_.
+In case of trouble, please check there if your issue has already been reported.
+If you spotted it first, help us smashing it by providing a detailed and welcomed feedback
+`here <https://github.com/OCA/stock-logistics-workflow/issues/new?body=module:%20stock_picking_invoice_link%0Aversion:%20{version}%0A%0A**Steps%20to%20reproduce**%0A-%20...%0A%0A**Current%20behavior**%0A%0A**Expected%20behavior**>`_.
```
Comment (eLBati) [inline stock_picking_invoice_link/README.rst]: :heavy_exclamation_mark: `{version}`

Comment (tafaRU) [inline stock_picking_invoice_link/README.rst]: Fixed, thanks!

File: stock_picking_invoice_link/README.rst [hunk 2]
```diff
@@ -0,0 +1,43 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :alt: License: AGPL-3
+
+Stock Picking Invoice Link
```
Comment (pedrobaeza) [inline stock_picking_invoice_link/README.rst]: Put a better title like "Link between the invoice and its source picking"

Comment (tafaRU) [inline stock_picking_invoice_link/README.rst]: Done, thanks!

File: stock_picking_invoice_link/stock.py [hunk 3]
```diff
@@ -18,69 +18,59 @@
...
 
 
-class stock_picking(orm.Model):
+class StockPicking(models.Model):
     _inherit = "stock.picking"
 
-    def _get_invoice_view_xmlid(self, cr, uid, ids, name, arg, context=None):
-        res = {}
-        for pick in self.browse(cr, uid, ids, context=context):
+    @api.multi
```
Comment (pedrobaeza) [inline stock_picking_invoice_link/stock.py]: Why not @api.one and you avoid a loop?

File: stock_picking_invoice_link/stock.py [hunk 4]
```diff
@@ -18,69 +18,59 @@
...
-            'stock.picking', 'invoice_id', 'Related Pickings', readonly=True,
-            copy=False,
-            help="Related pickings "
-            "(only when the invoice has been generated from the picking)."),
-    }
+    picking_ids = fields.One2many(
+        comodel_name='stock.picking', inverse_name='invoice_id',
+        string='Related Pickings', readonly=True,
+        copy=False,
+        help="Related pickings "
```
Comment (pedrobaeza) [inline stock_picking_invoice_link/stock.py:64]: Strange line split

File: stock_picking_invoice_link/stock.py [hunk 5]
```diff
@@ -18,69 +18,59 @@
...
-        'move_line_ids': fields.one2many(
-            'stock.move', 'invoice_line_id', 'Related Stock Moves',
-            readonly=True,
-            help="Related stock moves "
-            "(only when the invoice has been generated from the picking)."),
-    }
+    move_line_ids = fields.One2many(
+        comodel_name='stock.move', inverse_name='invoice_line_id',
+        string='Related Stock Moves', readonly=True,
+        help="Related stock moves "
```
Comment (pedrobaeza) [inline stock_picking_invoice_link/stock.py]: The same

File: stock_picking_invoice_link/stock.py (hunk 4 above)
Comment (tafaRU) [inline stock_picking_invoice_link/stock.py:64]: It was like that :expressionless: If you think it's better I could modify it.

File: stock_picking_invoice_link/stock.py (hunk 3 above)
Comment (tafaRU) [inline stock_picking_invoice_link/stock.py]: I agree, please see https://github.com/OCA/stock-logistics-workflow/commit/15e764212626c193a846fef54f4dc18713f82f12

File: stock_picking_invoice_link/stock.py (hunk 4 above)
Comment (pedrobaeza) [inline stock_picking_invoice_link/stock.py:64]: Well, I usually align the second line quote at the level of the first one, but PEP8 doesn't say anything about this, so what you want

Comment (tafaRU) [inline stock_picking_invoice_link/stock.py:64]: Me too usually! So https://github.com/OCA/stock-logistics-workflow/commit/57e6ac2c40108fe4f61b180a7ad3f32f49c8beae :wink: 

Does the pot file need to be exported again?

Comment (pedrobaeza) [inline stock_picking_invoice_link/stock.py:64]: No, because strings are concatenated

**Comment by pedrobaeza** [comment]
:+1:

**Comment by eLBati** [comment]
LGTM :+1:

**Comment by pedrobaeza** [comment]
Checking this again I see no interface buttons nor fields for showing the information picked.

**Comment by tafaRU** [comment]
@pedrobaeza, actually the only fields for showing the information picked are in the invoice form and in the picking form. What are you not convinced by?

**Comment by pedrobaeza** [comment]
I see no XML adding these fields in the views.

**Comment by tafaRU** [comment]
@pedrobaeza, XML files are contained here: https://github.com/tafaRU/stock-logistics-workflow/tree/8.0-stock_picking_invoice_link-imp/stock_picking_invoice_link They are not changed.

**Comment by pedrobaeza** [comment]
Sorry, you're right. This is only a migration to new API. I merge.

**Comment by tafaRU** [comment]
:wink:

**Pull request merged**